src/praisonai-agents/praisonaiagents/utils/async_bridge.py

```
@@ -0,0 +1,94 @@
+            new_loop = asyncio.new_event_loop()
+            try:
+                return new_loop.run_until_complete(coro)
+            finally:
+                new_loop.close()
+        
+        future = pool.submit(_run)
+        try:
+            return future.result(timeout=timeout)
+        except concurrent.futures.TimeoutError:
+            logger.error(f"Coroutine execution timed out after {timeout}s")
+            raise TimeoutError(f"Coroutine execution timed out after {timeout} seconds")
```

Ruff (0.15.10)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)